Frontiers_188_create_list_of_products_and_prices.py

```diff
@@ -0,0 +1,86 @@
+"""
+Setup:
+Must have a WordPress/Woocommerce website up and running. See the other homework.
+
+Problem Statement:
+Use woocommerce API Python library, to make API calls to list all products.
+Create a csv file that will have a list of products and their ‘price’ (there are 3 kinds of prices just use the field ‘price’).
+So the csv will have two columns ‘name’ and ‘price’.
+
+"""
+from woocommerce import API
+import csv
+
+wcapi = API(
+    url = "http://localhost:8888/mysite2",
+    consumer_key="<add consumer key>",
+    consumer_secret="<add consumer secret>",
+    version="wc/v3",
+    timeout=60
+)
+
+output_file = "product_list_of_prices_2.csv"
+product_list = []
+page = 1
+
+while True:
+    r = wcapi.get("products", params={"per_page": 100, 'page': page})
+    products = r.json()
+
+    status_code = r.status_code
+    if status_code != 200:
+        raise Exception(f"Expected status code 200. Got {status_code}")
```

Comment:
In your error message you should print the response body too. the status code will tell you there is an issue but the body will tell you what the issue is. So in the error message add 'r.content'.
You can add r.json() too but .content always prints info .json() assumes the response is json.

Reply:
This is handy, thanks. Is this what you had in mind?